manage.py

```diff
@@ -1,21 +1,153 @@
 #!/usr/bin/env python
 import os
+import re
+import socket
 import sys
+import time
+from typing import List
 
+import django
+import pyuwsgi
+from django.contrib.auth import get_user_model
+from django.core.management import call_command, execute_from_command_line
 
-# Separate definition to ease calling this in other scripts.
-def main():
+
+DEFAULT_ENVS = {
+    "DJANGO_SETTINGS_MODULE": "pydis_site.settings",
+    "SUPER_USERNAME": "admin",
+    "SUPER_PASSWORD": "admin",
+    "DEFAULT_BOT_API_KEY": "badbot13m0n8f570f942013fc818f234916ca531",
+}
+
+
+for key, value in DEFAULT_ENVS.items():
+    os.environ.setdefault(key, value)
+
+
+class SiteManager:
+    """
+    Manages the preparation and serving of the website.
+
+    Handles both development and production environments.
+
+    Usage:
+        manage.py run [option]...
+
+    Options:
+        --debug    Runs a development server with debug mode enabled.
+        --silent   Sets minimal console output.
+        --verbose  Sets verbose console output.
+    """
+
+    def __init__(self, args: List[str]):
+        self.debug = "--debug" in args
+        self.silent = "--silent" in args
+
+        if self.silent:
+            self.verbosity = 0
+        else:
+            self.verbosity = 2 if "--verbose" in args else 1
+
+        if self.verbosity and self.debug:
+            os.environ.setdefault("DEBUG", "true")
+            print("Starting in debug mode.")
+
+    @staticmethod
+    def create_superuser() -> None:
+        """Create a default django admin super user in development environments."""
+        print("Creating a superuser.")
+
+        name = os.environ["SUPER_USERNAME"]
+        password = os.environ["SUPER_PASSWORD"]
+        bot_token = os.environ["DEFAULT_BOT_API_KEY"]
+        user = get_user_model()
+
+        # Get or create admin superuser.
+        if user.objects.filter(username=name).exists():
+            user = user.objects.get(username=name)
+            print('Admin superuser already exists.')
+        else:
+            user = user.objects.create_superuser(name, '', password)
+            print('Admin superuser created.')
+
+        # Setup a default bot token to connect with site API
+        from rest_framework.authtoken.models import Token
+        token, is_new = Token.objects.update_or_create(user=user)
+        if token.key != bot_token:
+            token.delete()
+        token, is_new = Token.objects.update_or_create(user=user, key=bot_token)
+        if is_new:
+            print(f"New bot token created: {token}")
+        else:
+            print(f"Existing bot token found: {token}")
```

Comment:
Won't it replace the token if the existing token does not match? Does this really mean "existing bot token found but it's been replaced anyway"?

Reply:
`is_new` covers all cases where the default token doesn't match or doesn't exist in the first place.

`else` is only if the token actually is a match and truly wasn't recreated.